### PR TITLE
bpf: Coccinelle scripts for align_stack_8 and const qualifier

### PR DESCRIPTION
**Pull request by pchaigno**
Please see commit messages for details.

I'm planning to enable this in CI as a followup, if @qmonnet doesn't get to it first.

### CODE REVIEW

**Comment by pchaigno** [comment]
test-me-please

File: bpf/lib/nat46.h [hunk 1]
```diff
@@ -221,7 +221,8 @@ static __always_inline int ipv6_prefix_match(struct in6_addr *addr,
  * d6 = nat46_prefix<d4> or v6_dst if non null
  */
 static __always_inline int ipv4_to_ipv6(struct __ctx_buff *ctx, struct iphdr *ip4,
-					int nh_off, union v6addr *v6_dst)
+					int nh_off,
+					const union v6addr *v6_dst)
```
Comment (borkmann) [inline bpf/lib/nat46.h:225]: small nit: could go to prior line

File: bpf/include/bpf/ctx/xdp.h [hunk 2]
```diff
@@ -136,7 +136,8 @@ __csum_replace_by_4(__sum16 *sum, __wsum from, __wsum to)
 }
 
 static __always_inline __maybe_unused int
-l3_csum_replace(struct xdp_md *ctx, __u64 off, const __u32 from, __u32 to,
+l3_csum_replace(const struct xdp_md *ctx, __u64 off, const __u32 from,
+		__u32 to,
 		__u32 flags)
```
Comment (borkmann) [inline bpf/include/bpf/ctx/xdp.h:141]: small nit: could go to prior line

**Review by borkmann — APPROVED** [review]
Nice work, I like it! :-)

**Comment by borkmann** [comment]
We could probably also add a few others to the build assertion, e.g. https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/scripts/coccinelle has misc/array_size.cocci, misc/cond_no_effect.cocci and probably many others that might be relevant for us as well (but can be follow-up).

File: bpf/include/bpf/ctx/xdp.h (hunk 2 above)
Comment (pchaigno) [reply bpf/include/bpf/ctx/xdp.h:141]: :thinking: Not sure what happened here. Coccinelle is supposed to be good at handling this sort of things.

**Comment by pchaigno** [comment]
> We could probably also add a few others to the build assertion, e.g. https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/scripts/coccinelle has misc/array_size.cocci, misc/cond_no_effect.cocci and probably many others that might be relevant for us as well (but can be follow-up).

I think I'll create an issue to list those we might be interested in and add them in batch once coccicheck is enabled in CI.

@borkmann I've pushed an update and had to exclude cases `... when != e = x` in `const.cocci` to properly handle the following line: https://github.com/cilium/cilium/blob/97f8299d6c1a0da4a3f23c296f1e719c72e94c35/bpf/lib/nat.h#L314
That adds one more false negative, but I'd rather err on the side of no false positives if we want to enable it in CI afterward.

**Comment by pchaigno** [comment]
test-me-please

**Review by tklauser — APPROVED** [review]
Not too familiar with SmPL itself, so only briefly skimmed through. But the generated changes and the Makefile rule LGTM.

**Comment by pchaigno** [comment]
test-gke

**Comment by borkmann** [comment]
> > We could probably also add a few others to the build assertion, e.g. https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/scripts/coccinelle has misc/array_size.cocci, misc/cond_no_effect.cocci and probably many others that might be relevant for us as well (but can be follow-up).
> 
> I think I'll create an issue to list those we might be interested in and add them in batch once coccicheck is enabled in CI.

Sounds good to me, thanks!

**Comment by pchaigno** [comment]
@cilium/janitors I think the reviewer requests for `@cilium/ci` and `@cilium/janitors` are incorrect here and should be fixed by #11125.

**Review by qmonnet — APPROVED** [review]
Here's your Janitors' review anyway. Awesome work, thanks! Let's merge it.

**Pull request merged**